### DISCUSSION TITLE
Integrate with Intervals ICU - Open API to load workout of the day (event)

### DISCUSSION THREAD

**Discussion by Lunchtime0614** [cagnulein/qdomyos-zwift#1907] [Ideas]
Has there been any thought to integrate with https://intervals.icu which has an open API for users.  Something like pulling in the current workout for the current day so that QZ can show your currently planned workout and then execute it on your trainer?

https://intervals.icu/api/v1/docs/swagger-ui/index.html

Seems to have endpoints for getting a list of events on the calendar for a giving time period (lets say 1 week - the current week) and then from there an endpoint for reading each event on the calendar.  Also has an endpoint for downloading the event in Zwift/zwo format that could be used to load to QZ after a user picks which event they want to ride/train.

I'd be happy to help test, but I'm not skilled enough to build this...I can run API's etc but again not skilled enough to leverage the results.

**Comment by cagnulein** [comment]
Hi, it's definitely possible but it seems a lot of work :)
I will keep this in mind but I can't guarantee a timing :)

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 23 dic 2023 alle 01:58 Lunchtime0614 ***@***.***>
ha scritto:
